modules/local/dada2_quality.nf

```diff
@@ -13,12 +13,61 @@ process DADA2_QUALITY {
     output:
     path "${meta}_qual_stats.pdf"            , emit: pdf
     tuple val(meta), path("*_qual_stats.tsv"), emit: tsv
+    path "versions.yml"                      , emit: versions
     path "*.args.txt"                        , emit: args
+    path "*plotQualityProfile.txt"           , emit: warning
 
     script:
     def args = task.ext.args ?: ''
     """
-    dada_quality.r "${meta}_qual_stats" $args
-    echo 'plotQualityProfile\t$args' > "plotQualityProfile.args.txt"
+    #!/usr/bin/env Rscript
+
+    suppressPackageStartupMessages(library(dada2))
+    suppressPackageStartupMessages(library(ShortRead))
+
+    readfiles <- sort(list.files(".", pattern = ".fastq.gz", full.names = TRUE))
+
+    #make list of number of sequences
+    readfiles_length <- countLines(readfiles) / 4
+
+    #use only the first x files when read number gets above 2147483647, read numbers above that do not fit into an INT and crash the process!
+    if ( sum(readfiles_length) > 2147483647 ) {
```

Comment:
Put the numeric value in a vector (since you use it more often than once)

Reply:
Well, yes, I use it twice! Will do!